### ISSUE TITLE
[Bug]: When the query node is rolling upgraded, with 3 replicas, the first two query nodes are upgraded quickly, but the third query node waits for 30 minutes before completing the upgrade.

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36426]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.4.3--> master-20240920-eb23e23c-amd64
- Deployment mode(standalone or cluster):mixcoord
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

![image](https://github.com/user-attachments/assets/b66aa4f4-801d-4496-9313-772bff8c2832)

It can be inferred from the age of the query node that there was a 30-minute gap between the second query node being upgraded to the third query node being upgraded.

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/rolling_update_for_operator_test_simple/detail/rolling_update_for_operator_test_simple/4975/pipeline
log:
[artifacts-kafka-mixcoord-4975-server-logs.tar.gz](https://github.com/user-attachments/files/17092278/artifacts-kafka-mixcoord-4975-server-logs.tar.gz)

cluster: 4am
ns: chaos-testing
pod info
before upgrading
```
[2024-09-22T17:06:39.621Z] + kubectl get pods -o wide

[2024-09-22T17:06:39.623Z] + grep kafka-mixcoord-4975

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-etcd-0                                        1/1     Running       0                3m8s    10.104.23.130   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-etcd-1                                        1/1     Running       0                3m8s    10.104.18.18    4am-node25   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-etcd-2                                        1/1     Running       0                3m8s    10.104.34.116   4am-node37   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-0                                       2/2     Running       0                3m7s    10.104.23.133   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-1                                       2/2     Running       0                3m7s    10.104.34.118   4am-node37   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-2                                       2/2     Running       0                3m7s    10.104.18.21    4am-node25   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-exporter-578868bb6-qp47p                1/1     Running       3 (2m46s ago)    3m7s    10.104.14.121   4am-node18   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-zookeeper-0                             1/1     Running       0                3m7s    10.104.23.132   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-zookeeper-1                             1/1     Running       0                3m7s    10.104.25.150   4am-node30   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-kafka-zookeeper-2                             1/1     Running       0                3m7s    10.104.30.205   4am-node38   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-datanode-c989b6bc-5fk95                1/1     Running       0                117s    10.104.14.125   4am-node18   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-datanode-c989b6bc-9frjz                1/1     Running       0                117s    10.104.6.167    4am-node13   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-datanode-c989b6bc-mnkzp                1/1     Running       0                117s    10.104.23.137   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-indexnode-7df544f557-6h6rp             1/1     Running       0                117s    10.104.14.126   4am-node18   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-indexnode-7df544f557-7pd74             1/1     Running       0                117s    10.104.6.166    4am-node13   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-indexnode-7df544f557-dtlh4             1/1     Running       0                117s    10.104.23.136   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-mixcoord-5d5cdcf7bf-5g24n              1/1     Running       0                117s    10.104.6.168    4am-node13   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-proxy-9b74d69f5-qs4ml                  1/1     Running       0                117s    10.104.6.165    4am-node13   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-querynode-0-dfc85597d-ct4qq            1/1     Running       0                116s    10.104.14.127   4am-node18   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-querynode-0-dfc85597d-dqhbq            1/1     Running       0                116s    10.104.6.169    4am-node13   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-milvus-querynode-0-dfc85597d-t4x69            1/1     Running       0                116s    10.104.23.138   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-minio-0                                       1/1     Running       0                3m8s    10.104.23.131   4am-node27   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-minio-1                                       1/1     Running       0                3m8s    10.104.34.117   4am-node37   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-minio-2                                       1/1     Running       0                3m8s    10.104.25.151   4am-node30   <none>           <none>

[2024-09-22T17:06:39.878Z] kafka-mixcoord-4975-minio-3                                       1/1     Running       0                3m8s    10.104.18.19    4am-node25   <none>           <none>
```

after upgrading
```
[2024-09-22T18:06:13.678Z] + kubectl get pods -o wide

[2024-09-22T18:06:13.678Z] + grep kafka-mixcoord-4975

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-etcd-0                                       1/1     Running       0                62m     10.104.23.130   4am-node27   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-etcd-1                                       1/1     Running       0                62m     10.104.18.18    4am-node25   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-etcd-2                                       1/1     Running       0                62m     10.104.34.116   4am-node37   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-0                                      2/2     Running       0                62m     10.104.23.133   4am-node27   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-1                                      2/2     Running       0                62m     10.104.34.118   4am-node37   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-2                                      2/2     Running       0                62m     10.104.18.21    4am-node25   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-exporter-578868bb6-qp47p               1/1     Running       3 (62m ago)      62m     10.104.14.121   4am-node18   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-zookeeper-0                            1/1     Running       0                62m     10.104.23.132   4am-node27   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-zookeeper-1                            1/1     Running       0                62m     10.104.25.150   4am-node30   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-kafka-zookeeper-2                            1/1     Running       0                62m     10.104.30.205   4am-node38   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-datanode-787849f54f-59w8m             1/1     Running       0                8m42s   10.104.5.129    4am-node12   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-datanode-787849f54f-8s4xk             1/1     Running       0                8m      10.104.19.204   4am-node28   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-datanode-787849f54f-hscsj             1/1     Running       0                9m22s   10.104.16.67    4am-node21   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-indexnode-6fccf5b79-jgzbl             1/1     Running       0                53m     10.104.32.193   4am-node39   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-indexnode-6fccf5b79-nf86q             1/1     Running       0                52m     10.104.16.19    4am-node21   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-indexnode-6fccf5b79-sz7p5             1/1     Running       0                52m     10.104.21.214   4am-node24   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-mixcoord-bb675c875-nvv2n              1/1     Running       1 (48m ago)      48m     10.104.21.215   4am-node24   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-proxy-5957cd79fc-lcwzk                1/1     Running       0                4m59s   10.104.21.221   4am-node24   <none>           <none>

[2024-09-22T18:06:13.679Z] kafka-mixcoord-4975-milvus-querynode-1-576c9fc85-mzhcs           1/1     Running       0                44m     10.104.16.25    4am-node21   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-milvus-querynode-1-576c9fc85-p5m9r           1/1     Running       0                12m     10.104.5.124    4am-node12   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-milvus-querynode-1-576c9fc85-w2tkm           1/1     Running       0                43m     10.104.32.204   4am-node39   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-minio-0                                      1/1     Running       0                62m     10.104.23.131   4am-node27   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-minio-1                                      1/1     Running       0                62m     10.104.34.117   4am-node37   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-minio-2                                      1/1     Running       0                62m     10.104.25.151   4am-node30   <none>           <none>

[2024-09-22T18:06:13.680Z] kafka-mixcoord-4975-minio-3                                      1/1     Running       0                62m     10.104.18.19    4am-node25   <none>           <none>
```

**Comment by zhuwenxing** [comment]
/assign @weiliu1031 
PTAL

**Comment by weiliu1031** [comment]
please verify this with latest image

**Comment by zhuwenxing** [comment]
verified and fixed in `2.4.14` --> `master-20241105-b83b376c-amd64`

The difference in age between three querynodes is within 3 minutes.
```
[2024-11-05T06:42:04.369Z] + kubectl get pods -o wide

[2024-11-05T06:42:04.369Z] + grep kafka-mixcoord-5443

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-etcd-0                                        1/1     Running            0                31m     10.104.15.3     4am-node20   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-etcd-1                                        1/1     Running            0                31m     10.104.20.216   4am-node22   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-etcd-2                                        1/1     Running            0                31m     10.104.16.80    4am-node21   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-0                                       2/2     Running            2 (30m ago)      31m     10.104.20.219   4am-node22   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-1                                       2/2     Running            2 (30m ago)      31m     10.104.15.5     4am-node20   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-2                                       2/2     Running            2 (30m ago)      31m     10.104.16.83    4am-node21   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-exporter-6886f479cb-2rc9w               1/1     Running            5 (29m ago)      31m     10.104.20.213   4am-node22   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-zookeeper-0                             1/1     Running            0                31m     10.104.18.136   4am-node25   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-zookeeper-1                             1/1     Running            0                31m     10.104.20.220   4am-node22   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-kafka-zookeeper-2                             1/1     Running            0                31m     10.104.23.181   4am-node27   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-datanode-778cfd8cbb-2k96j              1/1     Running            0                8m18s   10.104.13.155   4am-node16   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-datanode-778cfd8cbb-drlvg              1/1     Running            0                8m59s   10.104.14.2     4am-node18   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-datanode-778cfd8cbb-f56fx              1/1     Running            0                7m27s   10.104.27.230   4am-node31   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-indexnode-8646dc7974-s7h95             1/1     Running            0                20m     10.104.18.172   4am-node25   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-indexnode-8646dc7974-sq799             1/1     Running            0                20m     10.104.26.93    4am-node32   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-indexnode-8646dc7974-wwrtx             1/1     Running            0                21m     10.104.9.243    4am-node14   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-mixcoord-d88bd7cf8-gvw2v               1/1     Running            0                16m     10.104.26.104   4am-node32   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-proxy-6875c79984-9d882                 1/1     Running            0                3m54s   10.104.9.40     4am-node14   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-querynode-1-7956fbf89d-dl8p7           1/1     Running            0                10m     10.104.4.89     4am-node11   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-querynode-1-7956fbf89d-x42gs           1/1     Running            0                13m     10.104.9.15     4am-node14   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-milvus-querynode-1-7956fbf89d-z9hzb           1/1     Running            0                12m     10.104.14.237   4am-node18   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-minio-0                                       1/1     Running            0                31m     10.104.15.4     4am-node20   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-minio-1                                       1/1     Running            0                31m     10.104.20.218   4am-node22   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-minio-2                                       1/1     Running            0                31m     10.104.18.135   4am-node25   <none>           <none>

[2024-11-05T06:42:04.625Z] kafka-mixcoord-5443-minio-3                                       1/1     Running            0                31m     10.104.26.64    4am-node32   <none>           <none>
```
